src/main/middleware/draftMiddleware.ts

```diff
@@ -0,0 +1,63 @@
+import * as express from 'express'
+import * as moment from 'moment'
+
+import { Draft } from '@hmcts/draft-store-client/dist/draft/draft'
+import { DraftService } from '@hmcts/draft-store-client/dist/draft/draftService'
+
+import { DraftDocument } from '../model/draftDocument'
+
+import { UUIDUtils } from '../utils/uuidUtils'
+
+/**
+ * Filters list of drafts to return only these matching external ID. If none of the drafts has external ID set
+ * then unchanged list is returned so that they can be migrated to new format with external ID (legacy drafts scenario).
+ */
+function tryFilterByExternalId<T extends DraftDocument> (drafts: Draft<T>[], externalId: string): Draft<T>[] {
+  if (drafts.filter(draft => draft.document.externalId !== undefined).length === 0) {
+    return drafts
+  }
+
+  return drafts.filter(item => item.document.externalId === externalId)
+}
+
+export class DraftMiddleware {
+
+  static requestHandler<T extends DraftDocument> (draftService: DraftService, draftType: string, limit: number, deserializeFn: (value: any) => T = (value) => value): express.RequestHandler {
```

Comment:
191 chars :)

Reply:
Yeah, type declarations take some space :)

Reply:
Split into several lines for readability

Reply:
Done